### PR TITLE
Fighting for runtime equality

### PR DESCRIPTION
**Pull request by ben-albrecht**
@ronawho made a cool plot of the distribution of correctness testing run times. The figure revealed that 99% of wall time is consumed by the top 1% of tests.

In this PR, we take back wall ~~street~~ clock times by addressing some of these slow tests. Below is a summary of how each test is dealt with:

* `ToggleSerialState.chpl`
    * `sleep()` now supports reals, so we divide our sleep time by `10.0`
* `arr-forall.chpl`
    * Increase memfraction
* `array-of-records-crash-1*`
    * This is a future testing a timeout bug, so we add a 10 second timeout 
        * when working, we should expect a run time of about 1.5 seconds
* `nbody`
   * Set correctness testing to `n=1000` for both consistency and reduced run time.
        * Updated `.good` file to reflect this changed input as well.
    * **Perf-team:** ( @lydia-duncan  @psahabu @benharsh @ronawho @bradcray ) - is this cool?

[*paratested*]

### CODE REVIEW

**Comment by ronawho** [comment]
Just FYI here's the 30 most expensive tests (according to last nights linux64 testing)

```sh
wget http://chapel-ci.us.cray.com/job/correctness-test-linux64/570/consoleText
grep -i "elapsed execution" consoleText | sed 's/.*\"\(.*\)\" - \(.*\) seconds]/\2\      \1/g' |sort -n |tail -n30
```

```
10.024	users/vass/domain_with_on-all
10.026	parallel/taskPar/figueroa/taskParallel
10.027	parallel/sync/waynew/record2c
10.041	modules/standard/Spawn/spawn-popen-wait
11.023	parallel/single/figueroa/ReadWriteMethods
12.684	reductions/vass/reductions-stress-mf
12.758	studies/amr/diffusion/level/Level_DiffusionBE_driver (compopts: 1)
13.034	exercises/boundedBuffer2
13.464	npb/is/diten/is_no_buckets (compopts: 1)
13.788	npb/is/diten/is (compopts: 1)
14.545	npb/is/diten/is_mt (compopts: 1)
16.142	studies/hpcc/HPL/stonea/serial/hplExample2
16.311	release/examples/primers/learnChapelInYMinutes (compopts: 1)
16.580	release/examples/benchmarks/miniMD/explicit/miniMD (compopts: 1)
20.917	reductions/vass/reductions-stress-fast (compopts: 1)
24.024	parallel/sync/figueroa/ReadMethods
24.024	parallel/sync/figueroa/WriteMethods
24.198	studies/shootout/nbody/sidelnik/nbody_vector_4
24.551	studies/shootout/nbody/sidelnik/nbody_rangesub_5
24.813	studies/amr/advection/amr/AMR_AdvectionCTU_driver (compopts: 1)
26.251	performance/bharshbarg/arr-forall
28.025	parallel/serial/figueroa/ToggleSerialState
30.422	parallel/forall/vass/ri-3-stress.not-fast
30.613	statements/lydia/forVersusWhilePerf (compopts: 1)
32.147	studies/shootout/nbody/sidelnik/nbody_iterator_7
32.440	studies/shootout/nbody/sidelnik/nbody_reductions_6
48.215	io/ferguson/ctests/qio_bits_test (compopts: 1)
62.041	io/ferguson/ctests/qio_test (compopts: 1)
300.023	users/vass/km/array-of-records-crash-1.explicit-type
300.027	users/vass/km/array-of-records-crash-1
```

**Comment by ronawho** [comment]
Also, I 100% support this PR. @vasslitvinov might be interested in .timeouts added for array-of-records-crash-1*. but I can't imagine you'll object to them

**Comment by lydia-duncan** [comment]
Left to my own devices, I'd have changed the n-body problem size to match one of the shootout sizes (500000, 5000000, 50000000, or 1000 as listed in the init file or on the description for checking if a program is correct, complete with expected output, here: http://benchmarksgame.alioth.debian.org/u64q/nbody-description.html#nbody)

**Comment by lydia-duncan** [comment]
Also, it should be noted that if we reduce long running tests, we reduce a variable of testing coverage - most interesting user programs should run longer than the tests you've listed here and it's possible that there are problems that only exhibit themselves in longer-running programs.  Therefore, by reducing the number of tests that go into the longer range, we are reducing our chances of seeing such potential problems.

**Comment by ben-albrecht** [comment]
@lydia-duncan 

* Would we be OK with `n=1000` for n-body then?
* While we'll be dropping correctness testing for the errors that might plague only long-running tests, we'll arguably still be testing them nightly with the performance tests, which I am not touching here. I am personally OK with that, if it significantly cuts down correctness test run times.

**Comment by ronawho** [comment]
+1 for using `n=1000` for nbody if that's an "official" problem size.

And while in some cases reducing the execution time can reduce test coverage, I don't think that's an issue for nbody or arr-forall, particularly since we do the full problem size with performance testing. I think this is just a case where the correctness size was too large, and this brings it more in line with the correctness execution times we see for the other shootouts.

**Comment by bradcray** [comment]
Me likey!

(and... interesting timing, I'd been wondering what a histogram like this would look like just the other day...)

I wonder if the suggested nbody problem size was bumped up from 500 to 1000 at some point on the shootout site.  It seems very odd that we'd choose a number as big as 500 yet so arbitrary (and this is the second time recently that I've had the impression that he may have changed these test run values -- thread-ring being the other).

I agree that these aren't the tests to count on for "stress testing" of long-running Chapel programs.  And I think if we did want a special stress test suite (separate from performance that is), it should be somewhere other than the correctness suite.

**Comment by ronawho** [comment]
@bradcray if you want to see the actual histograms: linux64 is [here](https://docs.google.com/spreadsheets/d/1iIUJtoc9qK6OZ78yXR3LvZnh8WlBJ294sMn6iPoiJJs/edit?usp=sharing) and a paratest run across chapcs* is [here](https://docs.google.com/spreadsheets/d/1Tel2ikM39sI9kp1WjsrsuNph4qNP4YwFx4IOYB_Di7s/edit?usp=sharing)

The paratest one is slightly cleaner and has more information

**Comment by lydia-duncan** [comment]
+1 for the n=1000, that seems reasonable to me

**Comment by lydia-duncan** [comment]
Also, I am convinced by the long-running test coverage argument, so this is fine by me.

**Comment by ben-albrecht** [comment]
OK - updated to `n=1000` - will merge when testing finishes (and passes)

File: test/users/vass/km/array-of-records-crash-1.explicit-type.timeout [hunk 1]
```diff
@@ -0,0 +1 @@
+10
```
Comment (vasslitvinov) [inline test/users/vass/km/array-of-records-crash-1.explicit-type.timeout]: +1
These two timeouts should have been there from the beginning, really.

**Pull request merged**